docs/proposals/multi-node/README.md

```diff
@@ -598,6 +609,24 @@ For example, if an ETCD cluster is scaled up from `3` to `5` and the new members
 
 Hence, the superfluous entries in both `members` array as well as the ETCD cluster need to be [cleaned up](#recommended-action-12) [as appropriate](#work-flows-only-on-the-leading-member).
 
+#### Member Leases
+
+One [Kubernetes `lease` object](https://kubernetes.io/docs/reference/kubernetes-api/cluster-resources/lease-v1/) per desired ETCD member is maintained by `etcd-druid` (preferrably, the `custodian` controller in `etcd-druid`).
+The `lease` objects will be created in the same `namespace` as their owning `Etcd` object and will have the same `name` as the member to which they correspond (which, in turn would be the same as [the `pod` name in which the member ETCD process runs](#member-name-as-the-key)).
+
+The `lease` objects are created and deleted only by `etcd-druid` but are continually renewed within the `leaseDurationSeconds` by the [individual `etcd-backup-restore` sidecars](#work-flows-independent-of-leader-election-in-all-members) (corresponding to their members) if the the corresponding ETCD member is ready and is part of the ETCD cluster.
+
+This will mean that `etcd-backup-restore` becomes Kubernetes-aware. But there are other reasons for making `etcd-backup-restore` Kubernetes-aware anyway (e.g. to [maintain health conditions](#health-check)).
+This enhancement should keep `etcd-backup-restore` backward compatible.
+But it should be possible to use `etcd-backup-restore` Kubernetes-unaware as before this proposal. This is possible either by auto-detecting the existence of kubeconfig or by an explicit command-line flag (such as `--enable-etcd-lease-renewal` which can be defaulted to `false` for backward compatibility).
+
+A `member` entry in the `Etcd` resource [`status`](#status) would be marked as `Ready` (with `reason: LeaseSucceeded`) if the corresponding `pod` is ready  and the corresponding `lease` has not yet expired.
+The `member` entry would be marked as `NotReady` if the corresponding `pod` is not ready (with reason `PodNotReady`) or as `Unknown` if the corresponding `lease` has expired (with `reason: LeaseExpired`).
+
+While renewing the lease, the `etcd-backup-restore` sidecars also maintain the ETCD member id in the `spec.holderIdentity` field of the corresponding `lease` object since this information is only available to the `ETCD` member processes and the `etcd-backup-restore` sidecars. When the `lease` objects are created by `etcd-druid`, the `spec.holderIdentity` field would be empty.
```

Comment:
I was under the assumption that the `role` field makes more sense to druid than the `id` field. For a human operator of the cluster, seeing the etcd members' roles in the `Etcd` status (like `Leader` and `Follower`) would make more sense from a debugging perspective, rather than seeing the etcd member ID. WDYT?

Reply:
Populating member `id` in `spec.holderIdentity` made natural sense. But populating role (`Leader` or `Follower`) didn't make sense because it is not identity as such.

I did think about populating both in `spec.holderIdentity`. For example,
```
spec:
  holderIdentity: 272e204152:Leader
```
If it is not a leader then the value might be either just the `id` or `id:` or `id:Follower`.

What do you think of that? Then we can support both `id` and `role`. To be honest, it looks kludgy 😅

Reply:
Yes, the whole concept of using `holderIdentity` for this seems a but weird. But then again, we would need _some_ way for backup sidecars to give this information to druid so that it may populate the status with information enough for a human operator to debug issues quickly. So in terms of utility, I'm fine with passing just the `Role` info (Leader or Follower), since the etcd member ID might not really be of much relevance at that point of time, especially since we're using the member name (ie, pod name) as the real "identifier" for the member etcds.

Reply:
👍  I don't like putting only `Leader` or `Follower` in `holderIdentity`. I will modify the proposal to include both member `id` and `role` in the `holderIdentity` (as in `272e204152:Leader`).

Reply:
Done in https://github.com/gardener/etcd-druid/pull/207/commits/cfe150ff0a2b76825b80893f84c153abba85b444.